stackslib/src/net/httpcore.rs

```diff
@@ -1026,22 +1062,34 @@ impl StacksHttp {
         }
     }
 
-    /// Register an API RPC endpoint
+    /// Register an API RPC endpoint.
+    /// Auto-generates a permissive regex for 400/404/405 detection
+    /// unless the handler provides its own via path_regex_permissive().
     pub fn register_rpc_endpoint<Handler: RPCRequestHandler + 'static>(
         &mut self,
         handler: Handler,
     ) {
+        let strict_regex = handler.path_regex();
+        let handler_permissive = handler.path_regex_permissive();
+
+        let permissive_regex = if handler_permissive.as_str() == strict_regex.as_str() {
+            make_permissive_regex(&strict_regex)
+        } else {
+            handler_permissive
+        };
```

Comment:
I'd suggest keeping the naming consistent unless there's a strong reason not to:

```suggestion
        let strict_regex = handler.path_regex();
        let permissive_regex = handler.path_regex_permissive();

        let permissive_regex = if permissive_regex.as_str() == strict_regex.as_str() {
            make_permissive_regex(&strict_regex)
        } else {
            permissive_regex
        };
```